react-tic-tac-toe/src/board.tsx

```diff
@@ -0,0 +1,196 @@
+                
+        return "";
+    }
+    
+    // check if 3 cells have same non-empty val - return the winner state; otherwise undefined 
+    private check3Cells(cells: CellValue[], pos0: number, pos1: number, pos2: number): GameState {
+        if (cells[pos0] === cells[pos1] &&
+            cells[pos1] === cells[pos2] &&
+            cells[pos0] !== "") {
+            if (cells[pos0] === "X") {
+                return "X Wins!";
+            }
+            return "O Wins!";
+        }
+        else {
+            return undefined;
+        }
+    }
+    
+    // list all empty cell positions
+    private findAllEmptyCells(cells : CellValue[]): number[] {
+        return cells.map((v, i) => { 
+            if (v === "") {
+                return i;
+            }
+            else { 
+                return undefined;
+            }
+        }).filter((v) => { return v !== undefined });        
+    }
+    
+    // make a move
+    private move(pos: number, val: CellValue, callback?: () => void): void {
+        if (this.state.gameState === "" &&
+            this.state.cells[pos] === "") {
+            let newCells = this.state.cells.slice();
+            newCells[pos] = val;
+            let oldState = this.state.gameState;
+            this.setState({cells: newCells, gameState: this.checkGameState(newCells, pos, val)}, ()=>{
+                if (this.state.gameState !== oldState) {
+                    this.handleGameStateChange(this.state.gameState);
+                }
+                if (callback) {        
+                    callback.call(this);
+                }
+            });                 
+        }
+    }
+    
+    // AI make a random move
+    private aiMove(): void {
+        let emptyCells = this.findAllEmptyCells(this.state.cells);
+        let pos = emptyCells[Math.floor(Math.random() * emptyCells.length)];
+        this.move(pos, aiCell);    
+    }
+    
+    // handle a new move from player
+    private handleNewPlayerMove(pos: number): void {
+        this.move(pos, playerCell, this.aiMove);
+    }
+   
+    render() {
+        var cells = this.state.cells.map((v, i) => {
+            return (
+                <Cell key={i} pos={i} val={v} handleMove={() => this.handleNewPlayerMove(i)} />
+            )           
+        } );
+        
+        return ( 
+            <div className="board"> 
+                {cells}
+            </div> 
+        )
+    }
+}
+
+interface CellProps extends React.Props<any> {
+    pos: number;
+    val: CellValue;   
+    handleMove: () => void;
+}
+
+class Cell extends React.Component<CellProps, void> {
+
+    // position of cell to className
+    private posToClassName(pos: number): string {
+        let className = "cell";
+        switch (Math.floor(pos / 3)) {
+            case 0: 
+                className += " top";
+                break;
+            case 2: 
+                className += " bottom";
+                break;
+            default: break;             
```

Comment:
Fixed in https://github.com/Microsoft/TypeScriptSamples/pull/86/commits/fbbea5d13b978fd3bfdb462458494be1391000a5